WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -403,6 +417,37 @@ private extension ShippingLabelFormViewModel {
             }
         }
 
+        // Insert customs row if necessary
+        if customsFormRequired && rows.firstIndex(where: { $0.type == .customs }) == nil {
+            guard let packageDetailsRow = rows.first(where: { $0.type == .packageDetails }),
+                  let packageDetailsRowIndex = rows.firstIndex(of: packageDetailsRow) else {
+                return
+            }
+
+            // Decide display mode for customs row based on whether package details has been validated
+            let customsRowState: ShippingLabelFormViewController.DisplayMode = packageDetailsRow.dataState == .pending ? .disabled : .editable
+            let customsRowIndex = rows.index(after: packageDetailsRowIndex)
+            let customsRow = Row(type: .customs, dataState: .pending, displayMode: customsRowState)
+            rows.insert(customsRow, at: customsRowIndex)
+        }
+
+        // Remove customs row if necessary
+        if !customsFormRequired,
+           let customsRowIndex = rows.firstIndex(where: { $0.type == .customs }) {
+            rows.remove(at: customsRowIndex)
+        }
```

Comment:
Good point! I have moved this part to a separate method that gets called from `updateRowsForCustomsIfNeeded` in f570f8131843ac5e204eac54cb5bba88a4464ba6, since it makes more sense that they belong there.